torchdynamo_poc/TorchDynamo Demo.ipynb

```diff
@@ -251,54 +285,89 @@
+      "File \u001b[0;32m~/torch-mlir/build/tools/torch-mlir/python_packages/torch_mlir/torch_mlir/compiler_utils.py:73\u001b[0m, in \u001b[0;36mrun_pipeline_with_repro_report\u001b[0;34m(module, pipeline, description)\u001b[0m\n\u001b[1;32m     64\u001b[0m     message \u001b[38;5;241m=\u001b[39m \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\"\"\u001b[39m\u001b[38;5;130;01m\\\u001b[39;00m\n\u001b[1;32m     65\u001b[0m \u001b[38;5;124m        \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mdescription\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m failed with the following diagnostics:\u001b[39m\n\u001b[1;32m     66\u001b[0m \u001b[38;5;124m        \u001b[39m\u001b[38;5;132;01m{\u001b[39;00msys\u001b[38;5;241m.\u001b[39mstderr\u001b[38;5;241m.\u001b[39mgetvalue()\u001b[38;5;132;01m}\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     70\u001b[0m \u001b[38;5;124m        Add \u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m{\u001b[39;00mdebug_options\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m to get the IR dump for debugging purpose.\u001b[39m\n\u001b[1;32m     71\u001b[0m \u001b[38;5;124m        \u001b[39m\u001b[38;5;124m\"\"\"\u001b[39m\n\u001b[1;32m     72\u001b[0m     trimmed_message \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;124m'\u001b[39m\u001b[38;5;241m.\u001b[39mjoin([m\u001b[38;5;241m.\u001b[39mlstrip() \u001b[38;5;28;01mfor\u001b[39;00m m \u001b[38;5;129;01min\u001b[39;00m message\u001b[38;5;241m.\u001b[39msplit(\u001b[38;5;124m'\u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;124m'\u001b[39m)])\n\u001b[0;32m---> 73\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m TorchMlirCompilerError(trimmed_message) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;28mNone\u001b[39m\n\u001b[1;32m     74\u001b[0m \u001b[38;5;28;01mfinally\u001b[39;00m:\n\u001b[1;32m     75\u001b[0m     sys\u001b[38;5;241m.\u001b[39mstderr \u001b[38;5;241m=\u001b[39m original_stderr\n",
+      "\u001b[0;31mTorchMlirCompilerError\u001b[0m: Lowering Torch Backend IR -> Linalg-on-Tensors Backend IR failed with the following diagnostics:\nerror: failed to legalize operation 'torch.constant.str'\nnote: see current operation: %0 = \"torch.constant.str\"() {value = \"Hello!\"} : () -> !torch.str\nerror: Module does not conform to the linalg-on-tensors backend contract. See dialect conversion legality information above.\n\n\nError can be reproduced with:\n$ torch-mlir-opt -pass-pipeline='torch-backend-to-linalg-on-tensors-backend-pipeline' /tmp/PrintStatementModule.mlir\nAdd '-mlir-print-ir-after-all -mlir-disable-threading' to get the IR dump for debugging purpose.\n"
```

Comment:
This error makes it seem like the problem is that we're using a string.  Is the error just misleading?  Would this fail with trying to print tensors too?

Reply:
Yeah, this error is misleading. Because the string op is not lowered to something, then the `torch.constant.str` input to the op stays in the IR when reaching the linalg-on-tensors backend, leading to the error. (The current error checking in torch-mlir just looks at the first invalid op and emits the error there)